sparse_strips/vello_common/src/flatten.rs

```
@@ -76,36 +76,46 @@ pub fn fill(path: &BezPath, affine: Affine, line_buf: &mut Vec<Line>) {
 
     let mut closed = false;
+    let mut is_nan = false;
 
-    kurbo::flatten(iter, TOL, |el| match el {
-        kurbo::PathEl::MoveTo(p) => {
-            if !closed && p0 != start {
-                close_path(start, p0, line_buf);
-            }
+    kurbo::flatten(iter, TOL, |el| {
+        is_nan |= el.is_nan();
 
-            closed = false;
-            start = p;
-            p0 = p;
-        }
-        kurbo::PathEl::LineTo(p) => {
-            let pt0 = Point::new(p0.x as f32, p0.y as f32);
-            let pt1 = Point::new(p.x as f32, p.y as f32);
-            line_buf.push(Line::new(pt0, pt1));
-            p0 = p;
-        }
-        el @ (kurbo::PathEl::QuadTo(_, _) | kurbo::PathEl::CurveTo(_, _, _)) => {
-            unreachable!("Path has been flattened, so shouldn't contain {el:?}.")
-        }
-        kurbo::PathEl::ClosePath => {
-            closed = true;
+        match el {
+            kurbo::PathEl::MoveTo(p) => {
+                if !closed && p0 != start {
+                    close_path(start, p0, line_buf);
+                }
 
-            close_path(start, p0, line_buf);
+                closed = false;
+                start = p;
+                p0 = p;
+            }
+            kurbo::PathEl::LineTo(p) => {
+                let pt0 = Point::new(p0.x as f32, p0.y as f32);
+                let pt1 = Point::new(p.x as f32, p.y as f32);
+                line_buf.push(Line::new(pt0, pt1));
+                p0 = p;
+            }
+            el @ (kurbo::PathEl::QuadTo(_, _) | kurbo::PathEl::CurveTo(_, _, _)) => {
+                unreachable!("Path has been flattened, so shouldn't contain {el:?}.")
+            }
+            kurbo::PathEl::ClosePath => {
+                closed = true;
+
+                close_path(start, p0, line_buf);
+            }
         }
     });
 
     if !closed {
         close_path(start, p0, line_buf);
     }
+
+    // A path that contains NaN is ill-defined, so ignore it.
+    if is_nan {
```

Would it make sense to log the presence of a NaN (maybe only in debug)? In this case, I think it would be helpful to log which exact value is NaN within the closure above?

---

I think that it would be reasonable. `log` is a reasonably small crate, silent failures are really unfriendly, and most reasonable programs will already be including log anyway.

One other thing which could be worth considering is keeping a source [`Location`](https://doc.rust-lang.org/stable/core/panic/struct.Location.html) (probably gated behind a feature or rustflag?). Because tracking down where `NaN` paths come from is a pain otherwise.

---

I think log makes more sense here. For downstream vello apps, it would make diagnosing performance issues easier.